core/src/components/checkbox/checkbox.ionic.scss

```diff
@@ -44,94 +56,340 @@
+ */
+:host(.checkbox-label-placement-start) .checkbox-wrapper {
+  flex-direction: row;
+}
+
+:host(.checkbox-label-placement-start) .label-text-wrapper {
+  /**
+   * The margin between the label and
+   * the checkbox should be on the end
+   * when the label sits at the start.
+   */
+  @include globals.margin(null, globals.$ionic-space-400, null, 0);
+}
+
+// Label Placement - End
+// ----------------------------------------------------------------
+
+/**
+ * Label is on the right of the checkbox in LTR and
+ * on the left in RTL.
+ */
+:host(.checkbox-label-placement-end) .checkbox-wrapper {
+  flex-direction: row-reverse;
+}
+
+/**
+ * The margin between the label and
+ * the checkbox should be on the start
+ * when the label sits at the end.
+ */
+:host(.checkbox-label-placement-end) .label-text-wrapper {
+  @include globals.margin(null, 0, null, globals.$ionic-space-400);
+}
+
+// Label Placement - Fixed
+// ----------------------------------------------------------------
+
+:host(.checkbox-label-placement-fixed) .label-text-wrapper {
+  /**
+   * The margin between the label and
+   * the checkbox should be on the end
+   * when the label sits at the start.
+   */
+  @include globals.margin(null, globals.$ionic-space-400, null, 0);
+}
+
+/**
+ * Label is on the left of the checkbox in LTR and
+ * on the right in RTL. Label also has a fixed width.
+ */
+:host(.checkbox-label-placement-fixed) .label-text-wrapper {
+  flex: 0 0 100px;
+
+  width: 100px;
+  min-width: 100px;
+  max-width: 200px;
+}
+
+// Label Placement - Stacked
+// ----------------------------------------------------------------
+
+/**
+ * Label is on top of the checkbox.
+ */
+:host(.checkbox-label-placement-stacked) .checkbox-wrapper {
+  flex-direction: column;
+}
+
+:host(.checkbox-label-placement-stacked) .label-text-wrapper {
+  @include globals.transform(scale(0.75));
+
+  /**
+   * The margin between the label and
+   * the checkbox should be on the bottom
+   * when the label sits at the top.
+   */
+  @include globals.margin(null, 0, globals.$ionic-space-400, 0);
+
+  /**
+   * Label text should not extend
+   * beyond the bounds of the checkbox.
+   */
+  max-width: calc(100% / 0.75);
+}
+
+// TODO(ROU-10796): uncomment this when the scss compilation issue is fixed
+// :host(.checkbox-label-placement-stacked.checkbox-alignment-start) .label-text-wrapper {
+//   @include globals.transform-origin(start, top);
+// }
+
+// :host(.checkbox-label-placement-stacked.checkbox-alignment-center) .label-text-wrapper {
+//   @include globals.transform-origin(center, top);
+// }
+
 // Ionic Design Checkbox Sizes
 // --------------------------------------------------
 :host(.checkbox-size-small) {
-  // Size
-  --size: #{$checkbox-ionic-small-size};
+  --border-radius: #{globals.$ionic-border-radius-050};
+  --size: #{globals.$ionic-scale-400};
+}
+
+:host(.checkbox-size-small.checkbox-shape-rectangular) {
+  --border-radius: #{globals.$ionic-border-radius-0};
+}
```

Comment:
This is not necessary because the border radius will always be 0 regardless of the size. So the border will be set correctly with just `.checkbox-shape-rectangular`.

```suggestion
```